CircleCI-Public/circleci-yaml-language-server
# feat: add resource classes and images for Windows & GPU executors

[Jira](https://circleci.atlassian.net/browse/LIFE-112)

# Description

This PR goes over many problems with autocompletion and validation of machine executors.

# Implementation details

* Remove obsolete `windows:` executors
* Add images and resource classes for many machine executors
* Update Xcode versions and macOS resource classes
* Do some refactoring along the way

# How to validate

There are many cases to validate when writing an executor. Try some valid and invalid cases and see if the error messages make sense.

----

For reference, this is the validation process I came up with for machine executors

![image](https://github.com/CircleCI-Public/circleci-yaml-language-server/assets/731497/afb94764-aeb0-4d46-ad65-a989f5c8c4ca)
